File: kotlin/Curso de Youtube/hello.kt
/*
    Usa lo mismo que java
    Java virtual machine
    No interactua con java, usa su principio
    Es de código abierto
    Evoluciona java, es iteroperable con java, puede interatuarse con las librerias de java
    Potencia los productos de la empresa
    No se preocupa por la migración
    Es el lenguaje principal para desarrollar aplicaciones en android
    No han caido en los ,is,os errores que java
    Es multiplataforma y multiparadigmatico
    Se adapta a los diferentes entornos
    Usos:  aplicaciones nativas en android, para crear back-end.
    Todo lo que funciona con java funciona con kotlin.
    Da soporte para crear webs,para crear aplicaiones plataformas, alicaiones de escritorio, aplicaiones de data science, estan en una etapa inicial, no estan del todo preparado.
    Se puede usar una misma base de código para ios android, se hace en partes diferentes la parte visual
    Valida data
    En ios, genera un códio c que le permita enender, optimiza todo, se tiene todo compilado.
    Se piuede decidir en que momenot se puede usar la multiplataforma.
    Es orientado a objetos.

*/

fun main(){
    println("Hola, kotlin")

    //  Fundamentos
    //      Variables
    var myString = "Esto es una cadena de texto"
        //myString = "aqui cqmbia el valor de la cadena de texto"
        //  myString = 6 - error
    println(myString)

    var myString2 : String = "Esto es otra cadena de texto"

    println(myString2)

    var myInt = 7
    myInt = myInt + 4
    println(myInt)
    println(myInt - 1)
    println(myInt)

    var myInt2 : Int = 5
    println(myInt2)

    var myDouble = 6.5
    println(myDouble)

    myDouble = 6.0
    println(myDouble)

    var myDouble2 : Double = 6.5
    var myFloat : Float = 6.5f
    var myBool : Boolean = true
    println(myFloat)
    println(myDouble2)
    println(myBool)

    //      Constantes
        //val myConst = "Mi propiedad constante"

    //  Estructura de control
    myInt = 11
    myString = "PedroDev"

    if (myInt == 10 && myString == "Hola"){
        println("El valor es 10")
    } else if (myInt == 11 || myString == "Hola"){
        println("El valor es 11")
    }else {
        println("El valor ni es 10 ni 11.")
    }

    //  Estructuras
    var myList = mutableListOf<String>("Pedro", "Aguilar", "Rodríguez")
    println(myList[1])
    myList.add("Pedro")
    println(myList)

    //      sets
    var mySet = setOf("Pedro", "Aguilar","Rodríguez","Pedro")
    println(mySet)

    //      Mapas
    //  Estructura que permite diferentes tipos
    var myMap = mutableMapOf("Pedro" to 36, "Aguilar" to 31, "Rodríguez" to 81)
    myMap["Damian"] = 80
    println(myMap["Pedro"])

    //  Bucles
    for (value in myList){
        println(value)
    }

    for (value in mySet){
        println(value)
    }

    for (value in myMap){
        println(value)
    }

    var myCounter = 0
    
    while(myCounter < myList.count()){
        println(myList[myCounter])
        myCounter ++
        myFunction()
    }

    //  opcionales
    var myOpcional : String? = null
    myOpcional = "Mi cadena de texto opcional"
    println(myOpcional)

    //  Funciones
    myFunction()

    //  Clases
    var MyClass = MyClass("Pedro",21)
    println(MyClass.age)
    MyClass.myFunction()

    //  Interpolación de datos
    println("Este es el valor de la variable myInt: $myInt")
}

//  Funciones
    fun myFunction(){
        println("Esto es una función")
    }

//  Clases 
class MyClass (val name : String, val age : Int){

    fun myFunction(){
        println("Esto es una función en una clase.")
    }

}

File: javascript/Curso de Youtube/hello.js
/*
    
    Fue concebido para la web.
    Se programa con html y css
    Si senecesitaba algo más }
    Funciona en el explorador de web
    Es el lenguaje de la web
    Entiende todos los exploradores web
    Java es diferente de JavaScript
    Java era la palabra de moda
    Es interpretado.
    Se basa en la sintaxis de Java
    ECMAScript. Es la especificación.
    Se emplea para el desarrollo front-end.
    JavaScript se puede ejecutar en el servidor.
    Se emplea frameworks
    Se puede emplear para hacer para aplicaiones de escritorio y moviles con herramientas especificas
    MongoDB esta montado sobre el lenguaje
    Imperativo
    Basado en prototipos
    Lenguaje de tipado fácil
    el Backend 
*/

console.log("Hola, JavaScript")

//  Variables

var myString = "Esto es una cadena de texto" // Var es global
console.log(myString)

let myString2 = "Esta es una cadena de texto" //  Scoup, donde se habre llave y donde se cierra
myString = "Aquí cambia el valor de la cadena de texto"
console.log(typeof myString2)

myString2 = 6
console.log(myString2)
console.log(typeof myString2)

let myNumber = 7
myNumber = myNumber + 4
console.log(myNumber)
console.log(myNumber - 1)
console.log(myString + " " + myNumber)

let myNumber2 = 6.5
console.log(myNumber2)
console.log(typeof myNumber2)

let myBool = false
myBool = true
console.log(myBool)
console.log(typeof myBool)

myBool = null
console.log(myBool + myNumber2)

myBool = undefined
console.log(myBool)

let myUndefined
console.log(myUndefined)

//  Constantes
const MY_CONST = "Mi propiedad constante"
    //  MY_CONST = "Otro valor" <-- Error
console.log(MY_CONST)

//myNumber = 10
myString = "Hola"

//  Estructura de control
if (myNumber == 10 && myString == "Hola"){

    console.log("El valor es 10")

} else if (myNumber == 11 || myString == "Hola") {

    console.log("El valor es 11")

}else {

    console.log("El valor no es 10")

}

myBool = null

if (myBool){
    console.log("Null")
}

//  Funciones
function myFunction(){
    //console.log("Mi funcion")
    return "Mi función"
}

console.log(myFunction())

//  Listas
let myList = ["Pedro", "Aguilar", "31","Rodriguez"]
console.log(myList)
console.log(myList[0])

//  Set
let mySet = new Set(["Pedro", "Aguilar", "31","Rodriguez"])
mySet.add("Pedro")
console.log(mySet)

//  Mapa
let myMap = new Map([["pedro", 23], ["Aguilar", 10], ["Rodríguez", 21]])
myMap.set("Damian", 20)
console.log(myMap)
console.log(myMap.get("Pedro"))

//  Bucles
for (const value of myList){
    
    console.log(value)

}

let myCounter = 0

while (myCounter < myList.length){
    console.log(myList[myCounter])
    myCounter++
}

//  Clases
class MyClass{
    constructor(name, age){
        this.name = name
        this.age = age
    }
}

let myClass = new MyClass("pedro", 21)
console.log(myClass)
console.log(myClass.name)

//  Enum
const MyEnum ={
    DART: "dart",
    PYTHON: "python",
    SWIFT: "swift",
    JAVA: "java",
    KOTLIN : "kotlin",
    JAVASCRIPT : "javascript"
}

const myEnum = MyEnum.JAVASCRIPT
console.log(myEnum)